cartographer/sensor/timed_point_cloud_data.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/sensor/timed_point_cloud_data.h"
+
+#include "cartographer/transform/proto/transform.pb.h"
+#include "cartographer/transform/transform.h"
+
+namespace cartographer {
+namespace sensor {
+
+proto::TimedPointCloudData ToProto(
+    const TimedPointCloudData& timed_point_cloud_data) {
+  proto::TimedPointCloudData proto;
+  proto.set_timestamp(common::ToUniversal(timed_point_cloud_data.time));
+  *proto.mutable_origin() = transform::ToProto(timed_point_cloud_data.origin);
+  for (const auto& range : timed_point_cloud_data.ranges) {
```

Comment:
Is using proto.mutable_point_data()->Reserve or a transform with back_inserter possible?

Reply:
back_inserter is not possible as RepeatedPtrField does not implement push_back, but I added a call to reserve.